tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/service/mapper/CurriculumMembershipMapperTest.java

```diff
@@ -0,0 +1,80 @@
+package com.transformuk.hee.tis.tcs.service.service.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.transformuk.hee.tis.tcs.api.dto.ConditionsOfJoiningDto;
+import com.transformuk.hee.tis.tcs.api.dto.ProgrammeMembershipDTO;
+import com.transformuk.hee.tis.tcs.api.enumeration.GoldGuideVersion;
+import com.transformuk.hee.tis.tcs.service.model.ConditionsOfJoining;
+import com.transformuk.hee.tis.tcs.service.model.CurriculumMembership;
+import com.transformuk.hee.tis.tcs.service.model.ProgrammeMembership;
+import com.transformuk.hee.tis.tcs.service.repository.ConditionsOfJoiningRepository;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.assertj.core.util.Lists;
+import org.assertj.core.util.Sets;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CurriculumMembershipMapperTest {
+
+  @Mock
+  ConditionsOfJoiningMapper conditionsOfJoiningMapperMock;
+
+  @Mock
+  ConditionsOfJoiningRepository conditionsOfJoiningRepositoryMock;
+
+  @InjectMocks
+  private CurriculumMembershipMapper testObj;
+
+  @Test
+  public void entityToDtoShouldReturnListOfAllElementsAsDto() {
+    UUID pmID = UUID.randomUUID();
+    UUID pm2ID = UUID.randomUUID();
+    ProgrammeMembership pm1 = new ProgrammeMembership(), pm2 = new ProgrammeMembership();
+    CurriculumMembership cm1 = new CurriculumMembership(), cm2 = new CurriculumMembership();
+    ConditionsOfJoining conditionsOfJoining = new ConditionsOfJoining();
+
+    pm1.setUuid(pmID);
+    pm2.setUuid(pm2ID);
+
+    cm1.setId(1L);
+    cm2.setId(2L);
+    cm1.setProgrammeMembership(pm1);
+    cm2.setProgrammeMembership(pm2);
+    pm1.setCurriculumMemberships(Sets.newLinkedHashSet(cm1));
+    pm2.setCurriculumMemberships(Sets.newLinkedHashSet(cm2));
+
+    conditionsOfJoining.setVersion(GoldGuideVersion.GG9);
+    conditionsOfJoining.setProgrammeMembershipUuid(pmID);
+    conditionsOfJoining.setSignedAt(Instant.now());
+    ConditionsOfJoiningMapper conditionsOfJoiningMapper = new ConditionsOfJoiningMapperImpl();
+    ConditionsOfJoiningDto conditionsOfJoiningDto
+        = conditionsOfJoiningMapper.toDto(conditionsOfJoining);
```

Comment:
The fields within `conditionsOfJoining` don't matter to the mapper;  you could create your own DTO & set the fields on it (or not even set the fields) to remove the `ConditionsOfJoiningMapperImpl`

Reply:
True. I think I was carelessly following-on from the fairly explicit setting of PM<->CM relationships to give some 'real' data, but you're right, its not necessary.